### PR TITLE
[6.x] Add whereNull and whereNotNull to Collection

### PR DESCRIPTION
**Pull request by SjorsO**
This PR adds `whereNull` and `whereNotNull` methods to the Collection class.

These methods are already available when building a query:
```php
$users = User::whereNotNull('email_verified_at')->get();
```

But as soon as you have a collection, you need to do this instead:
```php
$users = User::all();

$unverifiedUsers = $users->whereStrict('is_verified_at', null);

$verifiedUsers = $users->where('is_verified_at', '!==', null);
```

With this PR, you can do this instead:
```php
$users = User::all();

$unverifiedUsers = $users->whereNull('is_verified_at');

$verifiedUsers = $users->whereNotNull('is_verified_at');
```

### CODE REVIEW

File: src/Illuminate/Support/Traits/EnumeratesValues.php [hunk 1]
```diff
@@ -492,6 +492,28 @@ public function where($key, $operator = null, $value = null)
...
+
+    /**
+     * Filter items where the given key is null.
+     *
+     * @param  string  $key
+     * @return static
+     */
+    public function whereNotNull($key)
+    {
+        return $this->where($key, '!=', null);
```
Comment (derekmd) [inline src/Illuminate/Support/Traits/EnumeratesValues.php]: These methods would have to use a strict comparison operators `'==='` and `'!==`'.

These PHP comparisons are true: `'' == null`, `0 == null`, `0.0 == null`.

```php
Illuminate\Support\Collection::macro('whereNull', function ($key) {
  return $this->where($key, null);
});

Illuminate\Support\Collection::macro('whereNotNull', function ($key) {
    return $this->where($key, '!=', null);
});

$items = [
    ['falsy' => ''],
    ['falsy' => 0],
    ['falsy' => 0.0],
];

collect($items)->whereNull('falsy')->all();
// [
//   [
//     "falsy" => "",
//   ],
//   [
//     "falsy" => 0,
//   ],
//   [
//     "falsy" => 0.0,
//   ],
// ]

collect($items)->whereNotNull('falsy')->all();
// []
```

Comment (SjorsO) [reply src/Illuminate/Support/Traits/EnumeratesValues.php]: Good catch!

**Comment by taylorotwell** [comment]
What if you have a collection of only values? `collect([1, null, 3])`

File: src/Illuminate/Support/Traits/EnumeratesValues.php [hunk 2]
```diff
@@ -492,6 +492,28 @@ public function where($key, $operator = null, $value = null)
         return $this->filter($this->operatorForWhere(...func_get_args()));
     }
 
+    /**
+     * Filter items where the given key is not null.
+     *
+     * @param  string  $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Traits/EnumeratesValues.php]: `string|null`

File: src/Illuminate/Support/Traits/EnumeratesValues.php [hunk 3]
```diff
@@ -492,6 +492,28 @@ public function where($key, $operator = null, $value = null)
...
+     */
+    public function whereNull($key = null)
+    {
+        return $this->whereStrict($key, null);
+    }
+
+    /**
+     * Filter items where the given key is null.
+     *
+     * @param  string  $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Traits/EnumeratesValues.php]: `string|null`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by SjorsO** [comment]
@taylorotwell 

The documentation doesn't mention anything about using `->where()` on a collection of simple values, so I'm not sure if this is a good idea, but I just pushed a commit that adds support for `whereNull` and `whereNotNull` for collections of simple values.

File: tests/Support/SupportCollectionTest.php [hunk 4]
```diff
@@ -3908,6 +3908,76 @@ public function testGetWithNullReturnsNull($collection)
...
+    }
+
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testWhereNotNullWithoutKey($collection)
+    {
+        $data = new $collection([1, null, 3, 'null', false, true]);
+
+        $this->assertSame([
```
Comment (binaryk) [inline tests/Support/SupportCollectionTest.php]: Just a comment!

**Pull request merged**